app/models/ont/library.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Ont
+  # Ont::Library
+  class Library < ApplicationRecord
+    include Material
+
+    has_many :requests, foreign_key: :ont_library_id, inverse_of: :library, dependent: :destroy
```

Comment:
Does this dependent relationship mean that if you destroy the library you also destroy the request?  Is that what we want?

Reply:
I think this is what it does. I included this as on creating the library we modify the request, adding a tag to it. If we remove the library we still have a tagged request. Would a better option here be to include a join table from library to requests so you only delete the join table entry?

Reply:
Is the tag singular and specific to the library?  If so, maybe it should exist on the library and you can just set this to null out the relationship and not modify the requests.  The request can delegate to the library tag if needed.

Reply:
No, each well used to form a library has its requests tagged on library creation. All requests in a well will have the same additional tag

Reply:
OK sorry not sure if you've confirmed what I thought or not.  A library adds the same tag to each sample in the library?  So you could store that tag on the library instead of the request?

Reply:
No. A library's requests come from many wells. The requests in each well used to form a library are tagged with a different tag

Reply:
Ah yes OK so although all the samples in a well have the same library tag, the tag used is different for each well in the library.  OK.  Well maybe the library tag on a request should be stored in a different field on the request so that you can separate it from the tags it has that weren't imposed by the library.

Reply:
This is what we were discussing a few days ago when modelling out the relationships between Ont::Request and Ont::Library on Lucidcharts. Having multiple columns for tags was not desired

Reply:
I admit updating the Ont::Request with new tags is also not a great solution as it ties the request to the library. This is handled for Pacbio by a three-way join between library, request and tags. That may be a better approach here too

Reply:
I have made the change to move to using a join table in my [next branch](https://github.com/sanger/traction-service/tree/GPL-386-ont-library-graphql)